security/src/main/scala/pl/touk/nussknacker/ui/security/api/LoggedUser.scala

```diff
@@ -66,9 +85,16 @@ final case class CommonUser(
   }
 
   override val isAdmin: Boolean = false
+
+  override def canImpersonate: Boolean = globalPermissions
+    .map(_.toLowerCase)
+    .contains(Permission.OverrideUsername.toString.toLowerCase)
+
 }
 
 final case class AdminUser(id: String, username: String) extends LoggedUser {
   override def can(category: String, permission: Permission): Boolean = true
   override val isAdmin: Boolean                                       = true
+  override val impersonatedBy: Option[LoggedUser]                     = None
```

Comment:
Why admin user can't be impersonated? I thought that isAdmin is a kind of wildcard for user that have access to all permissions to all categories.

Reply:
you may say that admin rights are too powerful to allow for impersonation of such type of user. But if the admin is just a regular user with all permissions, maybe we can allow to do that. But let's check it first

Reply:
After some changes I made it possible to impersonate Admin user. 

I introduced a marker trait `RealLoggedUser` (Representing real users without impersonation: `CommonUser` and `AdminUser`) and created `ImpersonatedUser` having references to the impersonated and impersonating `RealLoggedUsers`.

Reply:
This structure is much better

Reply:
In the impersonation solutions, there is often a mechanism that tells if a given user can be impersonated. We don't have it yet (and maybe there is no need to have it). But IMO it is better not allowing to impersonate the admin role. Do you need it in your case? Maybe at the moment we don't have to do it?

Reply:
As I mentioned before, admin currently is only a user with all permissions to all categories. Our admin doesn't have more power than any other user. IMO we shouldn't add this distinction because we don't have such a requirement  at this moment and it will be confusing in the current roles model - if we want to have such a  distinction, we should add more clarifications in docs around users/roles handling.

Reply:
AFAIS the admin role is created by adding a proper flag to the user. I'm not sure if this is equal to the situation where the user has a full set of permissions. Then in the code, we explicitly check in some places if this is the admin role. That's why I see it as a security hole which is better to avoid if there is no use case to impersonate the admin.

Reply:
In the `BaseHttpService` class we have the `authorizeAdminUser` method. Do you think it should be replaced by e.g. `authorizerAllPermUser` or just removed? Because it's used in the context of two endpoints: config reload and GET config.

Reply:
After offline consultations we decided that there is an Admin user. I modified the documentation to include information about it.
After changes admin user cannot be impersonated by default. 
However i made it possible to impersonate an Admin by setting `isAdminImpersonationPossible` flag in the authorization configuration to `true`. I made it possible as some external systems working with Nussknacker need that option.